src/postprocess/chi0.jl

```diff
@@ -3,6 +3,12 @@ using IterativeSolvers
 using ForwardDiff
 using ProgressMeter
 
+struct FunctionPreconditioner
+    f_ldiv!
+end
+LinearAlgebra.ldiv!(y, P::FunctionPreconditioner, x) = P.f_ldiv!(y, x)
+LinearAlgebra.ldiv!(P::FunctionPreconditioner, x) = P.f_ldiv!(copy(x), x)
```

Comment:
Good catch

Reply:
Yeah but which? This kind of wrapper things is better put close to the point of use

Reply:
preconditioners.jl, which we move to the top level from `eigen` ?

Reply:
nah I don't think we should have this as a proper thing. It's just a kludge needed to communicate with the iterative solver because the whole ldiv/lmul is a bit of a mess. Might as well make it clear it's a workaround and have it just when we use it

Reply:
well my point kind of is: What if we also need it somewhere else? Defining it again with the same name is an issue and using yet another struct is a bit stupid, too. But maybe we should burn that bridge when we get there.

Reply:
yeah we'll see when we see the requirement of that hypothetical other usage

Reply:
I'm very bad at anticipating what future code will need so let's just not do it

Reply:
ok.